docs: Restructure flagd docs (#231)
open-feature/flagd: docs/configuration/configuration.md

  -k, --server-key-path string    Server side tls key path

>   -a, --sync-provider-args        Sync provider arguments as key values separated by =

Reviewer: Are sync provider args still used?

Author: They are not, however I was hesitant to remove them completely, viper will throw if an unexpected argument is passed, I can open an issue to follow the pattern of `sync-provider` where a warning is thrown? However we may be 'reintroducing' args in future development

Reviewer: We can address this issue [here](https://github.com/open-feature/flagd/issues/233).

  -d, --socket-path string        Set the flagd socket path.

>   -f, --uri strings               Set a sync provider uri to read data from this can be a filepath, url or reference to a kubernetes custom resource. Using multiple providers is supported, however if flag keys are duplicated across multiple sources it may lead to unexpected behavior.

Reviewer 2: ```suggestion
  -f, --uri strings               Set a sync provider uri to read data from this can be a filepath, url or reference to a kubernetes custom resource. Using multiple providers is supported, however if flag keys are duplicated across multiple sources the resulting behavior is undefined.
```

To use an existing FeatureFlagConfiguration custom resource, start flagD with the following command:

```shell
flagd start --uri core.openfeature.dev/default/my_example
```

> An additional optional flag `refreshtime` can be applied to shorten the cache refresh when using the Kubernetes provider ( The default is 5s ). As an example: 

Author: removed this section
